bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutation.java

```diff
@@ -108,78 +151,171 @@ MutateRowsRequest toRequest(){
      * {@link MutateRowsResponse} is complete.
      */
     void addCallback(ListenableFuture<MutateRowsResponse> bulkFuture) {
-      FutureCallback<MutateRowsResponse> callback = new FutureCallback<MutateRowsResponse>() {
-        @Override
-        public void onSuccess(MutateRowsResponse result) {
-          Iterator<Status> statuses = result.getStatusesList().iterator();
-          Iterator<SettableFuture<Empty>> entries = futures.iterator();
-          while (entries.hasNext() && statuses.hasNext()) {
-            SettableFuture<Empty> future = entries.next();
+      Futures.addCallback(
+          bulkFuture,
+          new FutureCallback<MutateRowsResponse>() {
+            @Override
+            public void onSuccess(MutateRowsResponse result) {
+              handleResult(result);
+            }
+
+            @Override
+            public void onFailure(Throwable t) {
+              setFailure(t);
+            }
+          });
+    }
+
+
+    private synchronized void handleResult(MutateRowsResponse result) {
+      if (this.requestManager == null) {
+        LOG.error("Got duplicate responses for bulk mutation.");
+        return;
+      }
+
+      long nextBackOff = BackOff.STOP;
+      try {
+        if(currentBackoff == null) {
+          currentBackoff = retryOptions.createBackoff();
+        }
+        nextBackOff = currentBackoff.nextBackOffMillis();
+      } catch (IOException e) {
```

Comment:
I don't think nextBackOffMillis will actually throw but a comment stating that we are ignoring on purpose would be helpful